### ISSUE TITLE
io.kestra.plugin.fs.smb.Upload overwrites folder with file

### ISSUE THREAD

**Issue by renemolenaar82** [kestra-io/kestra#5937]
### Describe the issue

When you define a task to upload a file like this:

```
  - id: upload
    type: io.kestra.plugin.fs.smb.Upload
    host: my-file-server
    port: "445"
    username: kestra
    password: kestra
    from: "{{ outputs.download_video.uri }}"
    to: "/Somefolder/Projects/destination"
```
It will delete a folder named "destination" and upload the file with filename "destination".

Technically, this is a user error, but it would be better if it would throw an error because there is already a folder that exists that matches the filename. Or perhaps, add a "overwrite: True" setting which is set to False by default.

I just had to restore 40GB of files because of this ;)

### Environment

- Kestra Version: 0.19.10

**Comment by Ben8t** [comment]
The `overwrite: True` seems a good idea here !

@renemolenaar82 what do you mean by "this is a user error" ? If the path is `to: /Somefolder/Projects/destination/` (added `/`) does it works as intended?

**Comment by renemolenaar82** [comment]
@Ben8t, what I meant is that my intention was to store a file defined under `outputs.download_video.uri` in a folder named `destination`.

By accident, I configured it like this:

to: "/Somefolder/Projects/destination"

Where Kestra expects "destination" to be a filename, not a folder. So that's an error on my part of not checking the documentation.

But still, there should be a check whether a folder or filename exists instead of just blindly overwriting it without warning.

**Comment by Ben8t** [comment]
Yes ok make total sense. Let's add `overwrite: False | True` (false default) and and throw warning when `overwrite: False` and the file already exist in the destination.

**Comment by renemolenaar82** [comment]
Sounds like a great idea.

**Comment by mgabelle** [comment]
Does this need a backport ? Or do we introduce it only from 0.22 ? @anna-geller

**Comment by Ben8t** [comment]
No need for backport afaik @mgabelle

**Comment by mgabelle** [comment]
Created a PR to solve this issue. The `overwrite` property will be available for all `Upload` tasks, not only smb.
